src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -2096,12 +2129,36 @@ private BoundExpression BindArgumentValue(DiagnosticBag diagnostics, ArgumentSyn
         private BoundExpression BindOutVariableArgument(DeclarationExpressionSyntax declarationExpression, DiagnosticBag diagnostics)
         {
             TypeSyntax typeSyntax = declarationExpression.Type;
-            var designation = (SingleVariableDesignationSyntax)declarationExpression.Designation;
+            VariableDesignationSyntax designation = declarationExpression.Designation;
+            switch (designation.Kind())
+            {
+                case SyntaxKind.DiscardedDesignation:
+                    {
+                        bool isVar;
+                        bool isConst = false;
+                        AliasSymbol alias;
+                        TypeSymbol declType = BindVariableType(designation, diagnostics, typeSyntax, ref isConst, out isVar, out alias);
```

Comment:
>BindVariableType [](start = 46, length = 16)

We do this binding over and over again, here and twice in the following method. It would be nice to do it once at the top of this method, and then pass the type (and designation) down.